python/test/unit/language/test_core.py

```diff
@@ -2263,17 +2275,45 @@ def test_scan_layouts(M, N, src_layout, axis, device):
     np.testing.assert_equal(z_ref, z_tri.cpu().numpy())
 
 
-layouts = [
-    BlockedLayout([1, 4], [8, 4], [4, 1], [1, 0], [1, 1], [1, 1], [0, 1]),
-    BlockedLayout([1, 4], [8, 4], [4, 1], [0, 1], [1, 1], [1, 1], [0, 1]),
-    BlockedLayout([4, 4], [2, 16], [4, 1], [1, 0], [1, 1], [1, 1], [0, 1]),
-    MmaLayout(version=(2, 0), warps_per_cta=[4, 1], ctas_per_cga=[1, 1], cta_split_num=[1, 1], cta_order=[0, 1],
-              instr_shape=[16, 8]),
-    MmaLayout(version=(2, 0), warps_per_cta=[2, 2], ctas_per_cga=[1, 1], cta_split_num=[1, 1], cta_order=[0, 1],
-              instr_shape=[16, 8]),
-    MmaLayout(version=(3, 0), warps_per_cta=[4, 1], ctas_per_cga=[1, 1], cta_split_num=[1, 1], cta_order=[1, 0],
-              instr_shape=[16, 16, 16]),
-]
+def get_amd_matrix_core_version(arch: str) -> int:
+    """ Determine matrix core type available on current GPU.
+        0 means no tensor cores are available
+        1 corresponds to MFMA in CDNA 1 architecture
+        2 corresponds to MFMA in CDNA 2 architecture
+        3 corresponds to MFMA in CDNA 3 architecture
+    """
+    if arch in ['gfx908']:
+        return 1
+    if arch in ['gfx90a']:
+        return 2
+    if arch in ['gfx940', 'gfx941', 'gfx942']:
+        return 3
+    return 0
+
+
+if is_hip():
+    matrix_core_version = get_amd_matrix_core_version(triton.runtime.driver.active.get_current_target()[1])
+    layouts = [
+        MfmaLayout(version=(matrix_core_version, 0), warps_per_cta=[4, 1], instr_shape=[32, 32], is_transposed=True),
+        MfmaLayout(version=(matrix_core_version, 0), warps_per_cta=[4, 1], instr_shape=[32, 32], is_transposed=False),
+        MfmaLayout(version=(matrix_core_version, 0), warps_per_cta=[4, 1], instr_shape=[16, 16], is_transposed=True),
+        MfmaLayout(version=(matrix_core_version, 0), warps_per_cta=[4, 1], instr_shape=[16, 16], is_transposed=False),
+        BlockedLayout([1, 1], [32, 2], [4, 1], [0, 1], [1, 1], [1, 1], [0, 1]),
+        BlockedLayout([1, 4], [16, 4], [4, 1], [0, 1], [1, 1], [1, 1], [0, 1]),
+        BlockedLayout([4, 4], [4, 16], [4, 1], [1, 0], [1, 1], [1, 1], [0, 1]),
+    ]
+else:
+    layouts = [
+        BlockedLayout([1, 4], [8, 4], [4, 1], [1, 0], [1, 1], [1, 1], [0, 1]),
+        BlockedLayout([1, 4], [8, 4], [4, 1], [0, 1], [1, 1], [1, 1], [0, 1]),
+        BlockedLayout([4, 4], [2, 16], [4, 1], [1, 0], [1, 1], [1, 1], [0, 1]),
```

Comment:
can we unify the block size cases? It would be good to make sure we have similar kind of coverage across targets (and when adding more tests when we catch bug to add it to all targets)

Reply:
hmm.. we need to make sure that threadsPerWarp parameter of BlockedLayout contains THREADS_PER_WARP threads. Something like this:
BlockedLayout([1, 4], [8, THREADS_PER_WARP/8], [4, 1], [1, 0], [1, 1], [1, 1], [0, 1])

What do you think about this?